### DISCUSSION TITLE
What is the AGPL v3 license?

### DISCUSSION THREAD

**Discussion by SahilKDas** [lichess-bot-devs/lichess-bot#1027] [Q&A] [answered]
I might sound like an idiot, but I looked everywhere and couldn't find a definition to help me understand the AGPL v3 license.

~~I won't admit I'm lazy~~ Thank You So Much!

**Comment by MarkZH** [comment]
You can find it on the main page by clicking on one of the tabs just below the file list. It's next to README, Security, and Code of Conduct. [Here's a direct link](https://github.com/lichess-bot-devs/lichess-bot?tab=AGPL-3.0-1-ov-file); scroll down below the file list after clicking.

**Reply by SahilKDas** [reply]
thank you, but my question was what does it mean?

**Reply by MarkZH** [reply, marked as answer]
It means that you can do whatever you want with the code in this repository, for example:
- Download a copy
- Run the code
- Sell the code
- Send the copied code to other people (for example, by making your own GitHub repository with the code)
- Modify the code
- Send the modified code to other people
- Sell the modified code

The only requirements are
- That the AGPLv3 license is included with all copies, including the modified ones. Modified versions are also under the AGPLv3.
- If you modify the code, there must be a prominent notice that the code has been modified and on what date.

**Reply by NiQ1** [reply]
> It means that you can do whatever you want with the code in this repository, for example:
> 
> * Download a copy
> * Run the code
> * Sell the code
> * Send the copied code to other people (for example, by making your own GitHub repository with the code)
> * Modify the code
> * Send the modified code to other people
> * Sell the modified code
> 
> The only requirements are
> 
> * That the AGPLv3 license is included with all copies, including the modified ones. Modified versions are also under the AGPLv3.
> * If you modify the code, there must be a prominent notice that the code has been modified and on what date.

The main difference between GPL and AGPL is that if you use AGPL code in a network server / web app, that also counts as distribution. In the case of lichess bots, it means that if you create a chess website of your own and decide to improve upon lichess bots and use them in your website then you have to disclose your presumably improved bots' source, even if you don't offer any binary downloads.
